### PR TITLE
Add support for building Python 3.11 wheels

### PR DESCRIPTION
**Pull request by raphaelreinauer**
Description:

 • Updated CIBW_BUILD to include cp311-* for Python 3.11.
 • Updated comments in wheels workflow

### CODE REVIEW

File: .github/workflows/wheels.yml [hunk 1]
```diff
@@ -20,20 +20,20 @@ jobs:
         env:
           # Specify which Python versions to build wheels
           # https://cibuildwheel.readthedocs.io/en/stable/options/#build-skip
-          CIBW_BUILD: "cp37-* cp38-* cp39-* cp310-*"
-          # Skip 32 bit architectures, musllinux, and i686, and macOS x86_64 wheels for CP3.8 -- CP3.10
-          CIBW_SKIP: "*-win32 *-musllinux_x86_64 *_i686 cp38-macosx_x86_64 cp39-macosx_x86_64 cp310-macosx_x86_64"
+          CIBW_BUILD: "cp37-* cp38-* cp39-* cp310-* cp311-*"
```
Comment (matteocao) [inline .github/workflows/wheels.yml:23]: @raphaelreinauer feel free to remove python 3.7 as well. It is anyway not supported anymore.

Comment (matteocao) [reply .github/workflows/wheels.yml:23]: Also, this will make the CI pass and we can merge your PR

Comment (raphaelreinauer) [reply .github/workflows/wheels.yml:23]: I'll do. Should I add Python 3.12 as well?

Comment (raphaelreinauer) [inline .github/workflows/wheels.yml:23]:
```suggestion
          CIBW_BUILD: "cp38-* cp39-* cp310-* cp311-*"
```

Comment (matteocao) [reply .github/workflows/wheels.yml:23]: let's try!

File: .github/workflows/wheels.yml [hunk 2]
```diff
@@ -20,20 +20,20 @@ jobs:
         env:
           # Specify which Python versions to build wheels
           # https://cibuildwheel.readthedocs.io/en/stable/options/#build-skip
-          CIBW_BUILD: "cp37-* cp38-* cp39-* cp310-*"
-          # Skip 32 bit architectures, musllinux, and i686, and macOS x86_64 wheels for CP3.8 -- CP3.10
-          CIBW_SKIP: "*-win32 *-musllinux_x86_64 *_i686 cp38-macosx_x86_64 cp39-macosx_x86_64 cp310-macosx_x86_64"
+          CIBW_BUILD: "cp38-* cp39-* cp310-* cp311-*  cp312-*"
+          # Skip 32 bit architectures, musllinux, and i686, and macOS x86_64 wheels for CP3.8 -- CP3.11
+          CIBW_SKIP: "*-win32 *-musllinux_x86_64 *_i686 cp38-macosx_x86_64 cp39-macosx_x86_64 cp310-macosx_x86_64 cp311-macosx_x86_64"
```
Comment (raphaelreinauer) [inline .github/workflows/wheels.yml:25]:
```suggestion
          # Skip 32 bit architectures, musllinux, and i686, and macOS x86_64 wheels for CP3.8 -- CP3.12
          CIBW_SKIP: "*-win32 *-musllinux_x86_64 *_i686 cp38-macosx_x86_64 cp39-macosx_x86_64 cp310-macosx_x86_64 cp311-macosx_x86_64 cp312-macosx_x86_64"
```

**Comment by matteocao** [comment]
@raphaelreinauer can you pelase also update the `ci.yml` file like so:

Screenshot 2024-05-29 at 22 35 27


Same for your PR on `giotto-ph`

**Pull request merged**